spec/features/home_page_spec.rb

```diff
@@ -92,4 +92,51 @@
       expect(find(@alert)).to have_content(@sign_in_msg)
     end
   end
+
+  describe 'contact us' do
+    before(:each) do
+      @name_textbox      = 'input#name'
+      @email_textbox     = 'input#email'
+      @select_menu       = 'select#subject'
+      @message_textarea  = 'textarea#message'
+      @message           = 'Hello, this is is cool.'
+      @send_button_text  = 'Send'
+      HydrusMailer.stub_chain(:contact_message, :deliver_now).and_return('a mailer')
+    end
+
+    it 'should show the full form and then generate an email when a user fills in the contact form when not logged in' do
+      logout
+      visit contact_path
+      expect(page).to have_selector(@name_textbox)
+      expect(page).to have_selector(@email_textbox)
+      expect(page).to have_selector(@select_menu)
+      expect(page).to have_selector(@message_textarea)
+      fill_in 'email', with: 'test@test.com'
+      fill_in 'message', with: @message
+      click_button @send_button_text
+      expect(page).to have_content 'Your message has been sent.'
+      expect(HydrusMailer).to have_received(:contact_message)
+    end
+
+    it 'should show an error message if the user does not enter a message to send' do
```

Comment:
This to me doesn't sound like a feature, and so I wouldn't write a feature test for it. Instead I'd write a controller test.  Controller tests are much faster to run.

Reply:
don't we need to have an integration test to have access to capybara, which i am using for these tests?  leaving as is